### PR TITLE
add budget to nwc table and controller

### PR DESCRIPTION
**Pull request by joshr4**

### CODE REVIEW

File: db/migrations/20231113220722_add_nwc_budget.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+exports.up = function (knex) {
+  return knex.schema.alterTable("wallet_connection", function (table) {
+    table.integer("budget");
```
Comment (joshr4) [inline db/migrations/20231113220722_add_nwc_budget.js:3]: should this be `notnullable`?

Comment (blastshielddown) [reply db/migrations/20231113220722_add_nwc_budget.js:3]: yes. and maybe we default it to the lowest msat budget amount?

Comment (blastshielddown) [reply db/migrations/20231113220722_add_nwc_budget.js:3]: also might want to add a positive check. some examples here: https://github.com/wavlake/catalog/blob/staging/db/migrations/20230210225257_add_positive_checks.js

Comment (joshr4) [reply db/migrations/20231113220722_add_nwc_budget.js]: updated 7ea144c8eab40483ccd4bff96fdb6b1d69bb79fd

File: db/migrations/20231115220722_add_nwc_budget.js [hunk 2]
```diff
@@ -2,12 +2,20 @@ exports.up = function (knex) {
   return knex.schema.alterTable("wallet_connection", function (table) {
     table.integer("budget").notNullable();
```
Comment (blastshielddown) [inline db/migrations/20231115220722_add_nwc_budget.js:3]: Should we add `msats` to the names of these columns so the units are explicit?

Comment (joshr4) [reply db/migrations/20231115220722_add_nwc_budget.js:3]: yeah we can do that, i was thinking sats, but msats is more future proof plus do we even store sats anywhere else?

Comment (blastshielddown) [reply db/migrations/20231115220722_add_nwc_budget.js:3]: no we always store in msats

**Pull request merged**